TACC-Cloud/hazmapper
# Task/WG-5: complete leaflet map

## Overview: ##

Besides mapillary/streetview work, this completes the leaflet map by adding:

* points 
  * clustering of marker points with marker-cluster
  * custom style with color and icon
  * unique marker for certain feature-types like point clouds, image video
* polygons/lines


There are some slight differences with angular version of hazmapper:
   * Style for features with an (e.g. images, videos etc) are basically the same except --global-color-accent--normal is used for color
   * Default style for points (that don't have an asset) is the same except the color is --global-color-accent--normal
   * Cluster marker is now the same size as feature markers (before it was smaller which felt counterintuitive as the cluster marker represents multiple markers)


The remaining things for leaflet map are the following:
 * support mapillary in our map [WG-392](https://tacc-main.atlassian.net/browse/WG-392)
 * improve zooming (issues also exist in the angular version) [WG-202](https://tacc-main.atlassian.net/browse/WG-202)
 
## PR Status: ##


## Related Jira tickets: ##

* [WG-5](https://tacc-main.atlassian.net/browse/WG-5)

## Summary of Changes: ##

## Testing Steps: ##
1. Test on http://localhost:4200/project/cd010f4d-3975-4fde-8bbd-b81ffb87273f (set `GeoapiBackendEnvironment.Staging` in `src/secret_local.ts`) 
2. Test on your own maps on Staging or anywhere else
3. see that rendering for points, point clouds, other things is the same
4. You could test with taggit-edited map project to see custom icons/colors

## UI Photos:

TODO

## Notes: ##

To check or do:
- [x] fix initial zoom when going to project list and then back to new map
- [x] use icons and style markers for different features.
- [x] ~~check zoom on clustered features behavior~~ out of scope improvement so will be post-react: Added better notes to https://tacc-main.atlassian.net/browse/WG-202
- [x] check custom style in feature.properties.style is being used
- [x] check if feature.style could also be queried and used (optional)
- [x] check if default style is correct
- [x] check if LARGE map renders quickly
- [x] regression in duplicate features in FeatureFIleTree (fixed in https://github.com/TACC-Cloud/hazmapper/pull/294)
